Volume/Supporting/Main.swift

```diff
@@ -38,25 +38,18 @@ struct Main: App {
     private func registerForPushNotifications() {
         UNUserNotificationCenter.current().getNotificationSettings { settings in
             if settings.authorizationStatus == .notDetermined {
-                UNUserNotificationCenter.current().requestAuthorization(options: [.alert, .badge, .sound]) { granted, error in
-                    if granted {
-                        DispatchQueue.main.async {
-                            UIApplication.shared.registerForRemoteNotifications()
-                        }
-                    }
-                }
-            } else if settings.authorizationStatus == .authorized {
-                DispatchQueue.main.async {
-                    UIApplication.shared.registerForRemoteNotifications()
-                }
+                UNUserNotificationCenter.current().requestAuthorization(options: [.alert, .badge, .sound]) { _, _ in }
             }
         }
+        DispatchQueue.main.async {
+            UIApplication.shared.registerForRemoteNotifications()
+        }
```

Comment:
Actually, this method is called on every app launch. I verified that remote notifications will be registered on the first launch, as well later launches if the first few were attempted offline. The primary reason I moved this call outside the authorization block is because I want to register for remote notifications regardless of if the user would like to receive push notifications; the user simply won't see them, but we can still issue background updates. On Volume we use `deviceToken` as part of the user registration flow on backend, and remote notifications are the only way to receive device tokens.

Reply:
I still don't see how this works. User downloads app and opens for the first time. Notifs are not allowed, so we ask for permission. This is non blocking. The function keeps executing. We are not registered for notifs, so we attempt to register. This fails. Then the user finally clicks the button to allow notifs. Will we now send the device token to the server?

Reply:
Registering for remote notifications doesn't depend on user approval for push notifications. APNs can send silent remote notifications that don't appear as push notifications, which is what we want.

Reply:
Ok sounds good. Thanks for explaining.